packages/webpack-plugin/src/index.js

```diff
@@ -101,6 +109,23 @@ class LoadablePlugin {
   apply(compiler) {
     this.compiler = compiler
 
+    if (
+      this.opts.serverSideModuleFederation === true &&
+      (compiler.options.target === 'web' ||
+        compiler.options.target === undefined)
+    ) {
+      throw new Error(
+        `Enabling server-side module federation support for a client-side Webpack target (${compiler.options.target}) is unnecessary and will effectively disable all code-splitting.`,
+      )
+    }
+
+    if (this.opts.serverSideModuleFederation) {
+      Object.defineProperty(process, 'serverSideModuleFederation', {
```

Comment:
👍 that's a fancy way to "teleport" settings.
However, the particular way you did it will break some _assumptions_ of node creating a hard to understand "gap" for a few (rare) users of thread-loader or similar stuff.

Ease to fix. Just store what is required in [process.env](https://github.com/JaneaSystems/node/blob/2df82581162337f9e402d55f2d2d87585cb81dea/lib/child_process.js#L518) - a special place _designed_ for this.